### PR TITLE
[prometheus-node-exporter] allow configuration of ipFamilyPolicy

### PR DESCRIPTION
**Pull request by dongjiang1989**
#### What this PR does / why we need it

Allow configuration of ipFamilyPolicy

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/prometheus-node-exporter/templates/service.yaml [hunk 1]
```diff
@@ -10,6 +10,11 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+{{- if ge (.Capabilities.KubeVersion.Minor | int) 20 }}
+  ipFamilies:
+{{ toYaml .Values.service.ipFamilies | indent 2 }}
+  ipFamilyPolicy: {{ .Values.service.ipFamilyPolicy }}
+{{- end }}
```
Comment (zeritti) [inline charts/prometheus-node-exporter/templates/service.yaml:17]: Dual-stack services were made available as alpha in release [1.20](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.20.md) behind feature gate `IPv6DualStack`, disabled by default. The condition may not therefore be reliable. The feature became beta in 1.21 and stable in 1.23.

What about a toggle? One would not need to bother with dual stack making thus no change unless it is enabled. For example:

```
spec:
  {{- if .Values.service.ipDualStack.enabled }}
  ipFamilies: {{ toYaml .Values.service.ipDualStack.ipFamilies | nindent 4 }}
  ipFamilyPolicy: {{ .Values.service.ipDualStack.ipFamilyPolicy }}
  {{- end }}
```

File: charts/prometheus-node-exporter/values.yaml [hunk 2]
```diff
@@ -73,6 +73,9 @@ service:
   listenOnAllInterfaces: true
   annotations:
     prometheus.io/scrape: "true"
+  ipFamilies:
+  - IPv4
+  ipFamilyPolicy: SingleStack
```
Comment (zeritti) [inline charts/prometheus-node-exporter/values.yaml:78]: Further on the comment above:

```
service:
  ipDualStack:
    enabled: false
    ipFamilies: ["IPv6", "IPv4"]
    ipFamilyPolicy: "PreferDualStack"
```

File: charts/prometheus-node-exporter/templates/service.yaml (hunk 1 above)
Comment (dongjiang1989) [reply charts/prometheus-node-exporter/templates/service.yaml]: Great! Done. Add the `flag` for DualStack

File: charts/prometheus-node-exporter/values.yaml (hunk 2 above)
Comment (dongjiang1989) [reply charts/prometheus-node-exporter/values.yaml]: Done

**Comment by dongjiang1989** [comment]
@zeritti Please re-check it

File: charts/prometheus-node-exporter/Chart.yaml [hunk 3]
```diff
@@ -6,7 +6,7 @@ keywords:
   - prometheus
   - exporter
 type: application
-version: 4.20.1
+version: 4.20.2
```
Comment (zeritti) [inline charts/prometheus-node-exporter/Chart.yaml:9]:
```suggestion
version: 4.21.0
```

Please, raise the minor number with a new feature.

**Comment by zeritti** [comment]
Thank you, @dongjiang1989, lgtm.

**Review by zeritti — APPROVED** [review]

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**